nitrates/analysis_seeds/UpperLimits.py

```diff
@@ -0,0 +1,308 @@
+    )
+    parser.add_argument(
+        "--api_token",
+        type=str,
+        help="EchoAPI key for interactions.",
+        default=None
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+
+
+
+
+
+def main(args):
+    logging.basicConfig(
+        filename="upper_limits_analysis.log",
+        level=logging.DEBUG,
+        format="%(asctime)s-" "%(levelname)s- %(message)s",
+    )
+
+    work_dir=args.work_dir
+    resp_dname=args.resp_dname
+   # work_dir='/storage/home/gzr5209/Desktop/709523649_c0/'
+
+    conn = get_conn(os.path.join(work_dir,'results.db'))
+    logging.info("Connecting to DB") 
+
+    info_tab = get_info_tab(conn)
+    logging.info("Got info table")
+
+    trigger_time = info_tab['trigtimeMET'][0]
+    logging.debug("trigtime: %.3f" % (trigger_time))    
+    # default file names made by do_data_setup.py
+    evfname = os.path.join(work_dir,'filter_evdata.fits')
+    ev_data = fits.open(evfname)[1].data
+    # GTI extensions added to event fits file by do_data_setup.py
+    GTI_PNT = Table.read(evfname, hdu='GTI_POINTING') # when the analysis can be run (good data time and pointing)
+    GTI_SLEW = Table.read(evfname, hdu='GTI_SLEW') # slewing times
+    # the final set of detectors to mask (disabled dets, hot/cold dets, and dets with glitches)  
+    dmask = fits.open(os.path.join(work_dir,'detmask.fits'))[0].data
+    attfile = fits.open(os.path.join(work_dir,'attitude.fits'))[1].data
+    att_q = attfile["QPARAM"][np.argmin(np.abs(attfile["TIME"] - trigger_time))]
+    ra1, dec1 = convert_theta_phi2radec(100,100,att_q)
+    print('test ra, dec=',ra1,dec1)
+
+    ebins0 = np.array([15.0, 24.0, 35.0, 48.0, 64.0])
+    ebins0 = np.append(ebins0, np.logspace(np.log10(84.0), np.log10(500.0), 5+1))[:-1]
+    ebins0 = np.round(ebins0, decimals=1)[:-1]
+    ebins1 = np.append(ebins0[1:], [350.0])
+    nebins = len(ebins0)
+    logging.debug("Number of ebins: ")
+    logging.debug(nebins)
+
+    # number of detectors being used
+    ndets = np.sum(dmask==0)
+    logging.debug("Ndets: ")
+    logging.debug(np.sum(dmask==0))
+
+    t_end = trigger_time + 1e3
+    t_start = trigger_time - 1e3
+    mask_vals = mask_detxy(dmask, ev_data)
+
+    logging.debug(mask_vals)
+    bl_dmask = (dmask==0.)
+
+    # get rid of events:
+    # far away from trigger time
+    # from bad dets
+    # with bad event flags
+    bl_ev = (ev_data['EVENT_FLAGS']<1)&\
+        (ev_data['ENERGY']<=500.)&(ev_data['ENERGY']>=14.)&\
+        (mask_vals==0.)&(ev_data['TIME']<=t_end)&\
+        (ev_data['TIME']>=t_start)
+
+    #logging.info("Number of events passing cuts: ", np.sum(bl_ev))
+    ev_data0 = ev_data[bl_ev]
+
+   # logging.info(GTI_PNT)
+
+   # for row in GTI_PNT:
+   #     print(row['START'] - trigger_time, row['STOP'] - trigger_time)
+
+        
+    bkg_fname = os.path.join(work_dir,'bkg_estimation.csv')
+    bkg_df = pd.read_csv(bkg_fname)
+
+
+# create bkg object
+
+    try:
```

Comment:
Looks like you're not using GTI, just GTI_PNT so you can remove this try except statement